mteb/tasks/Retrieval/fas/NeuCLIR2022FasRetrieval.py

```diff
@@ -0,0 +1,34 @@
+from __future__ import annotations
+
+from mteb.abstasks.TaskMetadata import TaskMetadata
+
+from ....abstasks.AbsTaskRetrieval import AbsTaskRetrieval
+
+
+class NeuCLIR2022FasRetrieval(AbsTaskRetrieval):
+    metadata = TaskMetadata(
+        name="NeuCLIRFas2022",
+        description="The task involves identifying and retrieving the documents that are relevant to the queries.",
+        reference="https://neuclir.github.io/",
+        dataset={
+            "path": "mteb/neuclir-2022-fas",
+            "revision": "fa3b9068b9318d13ed5f065d9bbd5c8aff2dcfec",
+        },
+        type="Retrieval",
+        category="s2p",
+        eval_splits=["test"],
+        eval_langs=["fas-Arab"],
+        main_score="ndcg_at_20",
+        date=None,
+        form=None,
+        domains=None,
+        task_subtypes=None,
+        license=None,
+        socioeconomic_status=None,
+        annotations_creators=None,
+        dialect=None,
+        text_creation=None,
+        bibtex_citation=None,
+        n_samples=None,
+        avg_character_length=None,
```

Comment:
These will have to be specified as well. Let me know if you have any question related to these.

Reply:
It seems like most of these aren't filled out in the other datasets so I left them blank.. Many of them I don't have the answer for either. I will add as many as I know!

Reply:
Do you have a reference to what some of these are?

`annotations_creators` could be qrels or documents or queries?
Not sure what socioeconomic status indicates?

Reply:
> It seems like most of these aren't filled out in the other datasets so I left them blank.. Many of them I don't have the answer for either. I will add as many as I know!

You are completely correct, older datasets are missing their metadata, but new ones should have it.

> Do you have a reference to what some of these are?

You can find a description of the metadata object [here](https://github.com/embeddings-benchmark/mteb/blob/main/docs/adding_a_dataset.md#2-creating-the-metadata-object). Feel free to let me know if the documentation is missing.

> annotations_creators could be qrels or documents or queries? Not sure what socioeconomic status indicates?

Socioeconomic status for e.g. legal documents is "high" (lawyers are typically quite well of), the same goes for text by similar socio-demographic groups (politicians, doctors etc.). Web is often "mixed". The annotation is there as many NLP resources rarely targets low-income groups.

Annotation creators is who generated the annotations (e.g. expert annotated, found, etc.)

Reply:
Thanks, this was very helpful! 

Quick question on the `avg_character_length` and `n_samples` fields, are they the docs/queries/qrels?

Reply:
Ah yes that is a bit unclear atm.: query-document pairs (updated the taskmetadata docstring to also reflect this: #353)

Reply:
Thanks! I see how for different tasks it can be different.  

Just to confirm, for retrieval you mean the qrels (the query-document relevance assessments, which can be > 1 per query)?  Does that include non-relevant judgements or only the relevant judgements?

For avg_character_length, I am unclear on what this means ("query-document" pairs meaning the concatenation of the query with each relevant document)?

Reply:
I have previously done it by combining them into one pile and calculating the mean, but the concatenation of the two might be easier to interpret. We should probably create a method on the AbstaskRetrieval to make it consistent. Something you might want to @orionw?

Reply:
What do you mean by "create a method on the AbstaskRetrieval"?  Like a literal code method or just a way to do it?

I would suggest probably making it optionally a tuple, of (length query, length document, # of relevance judgements) as that's usually all the descriptive stats people want.  Or if you want it simpler, perhaps just average query + doc length for relevant docs.

Reply:
> Like a literal code method or just a way to do it?

Yes something like:

```
MyRetrievalTask().get_length()
MyRetrievalTask().get_n_samples()
```

 > Or if you want it simpler, perhaps just average query + doc length for relevant docs.
 
I would go for this. For nuanced metrics can always be calculated from the data (this is mostly to get is an overview of the size of the datasets)

Reply:
Okay added them. Thanks!